### PR TITLE
Consolidate and migrate kyverno to TF

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: images/kyverno/main.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+module "version-tags" {
+  for_each = local.components
+  source   = "../../tflib/version-tags"
+
+  package = local.packages[each.key]
+  config  = module.latest[each.key].config
+}
+
+module "test-latest" {
```
Comment (imjasonh) [inline images/kyverno/main.tf:47]: No dev variants?

File: images/kyverno/main.tf [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    "init" : "kyverno-init-container",
+  })
+
+  repositories = merge({
+    for k, v in local.components : k => "${var.target_repository}-${k}-controller"
+    }, {
+    "admission" : var.target_repository,
+    "cli" : "${var.target_repository}-cli",
+    "init" : "${var.target_repository}pre",
+  })
```
Comment (imjasonh) [inline images/kyverno/main.tf:24]:
```suggestion
  // Normally this is cgr.dev/chainguard/kyverno-cleanup-controller
  // But some images are named differently:
  // - cli -> cgr.dev/chainguard/kyverno-cli
  // - init -> cgr.dev/chainguard/kyvernopre
  repositories = merge({
    for k, v in local.components : k => "${var.target_repository}-${k}-controller"
    }, {
    "admission" : var.target_repository,
    "cli" : "${var.target_repository}-cli",
    "init" : "${var.target_repository}pre",
  })
```

File: images/kyverno/main.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+locals {
+  components = toset(["admission", "background", "cleanup", "cli", "init", "reports"])
+
+  packages = merge({
+    for k, v in local.components : k => "kyverno-${k}-controller"
+    }, {
+    "admission" : "kyverno",
+    "cli" : "kyverno-cli",
+    "init" : "kyverno-init-container",
+  })
```
Comment (imjasonh) [inline images/kyverno/main.tf:16]:
```suggestion
  // Normally the package is named like "kyverno-cleanup-controller"
  // But some packages are named differently:
  // - admission -> kyverno
  // - kyverno -> kyverno-cli
  // - init -> kyverno-init-container
  packages = merge({
    for k, v in local.components : k => "kyverno-${k}-controller"
    }, {
    "admission" : "kyverno",
    "cli" : "kyverno-cli",
    "init" : "kyverno-init-container",
  })
```

File: images/kyverno/main.tf (hunk 1 above)
Comment (mattmoor) [reply images/kyverno/main.tf:47]: it doesn't have them currently, so I'm not adding them

**Review by imjasonh — APPROVED** [review]

**Pull request merged**